modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -5262,6 +5281,37 @@ private void _updateResourcePermissions(
 			String.valueOf(objectEntry.getObjectEntryId()), modelPermissions);
 	}
 
+	private void _updateRootDescendantNodeObjectEntryStatus(
+			long userId, ObjectEntry objectEntry, ServiceContext serviceContext)
+		throws PortalException {
+
+		boolean skipModelListener = _skipModelListeners.get();
+		boolean skipObjectActionExecution =
+			ObjectActionThreadLocal.isSkipObjectActionExecution();
+
+		try {
+			_skipModelListeners.set(true);
+
+			ObjectActionThreadLocal.setSkipObjectActionExecution(true);
+
+			for (ObjectEntry rootDescendantNodeObjectEntry :
```

Comment:
Do you think there might be a large amount of entries here that would require implementing this with performance in mind? Like using `ActionableDynamicQuery` or something like that.

Reply:
@guilhermedcamacho last time I tried to use `ActionableDynamicQuery` with performance in mind, Brian asked me to implement a finder, see [here](https://github.com/brianchandotcom/liferay-portal/pull/157903/commits/49833779e24c7895a62da464ae4aec1e632a9345) and [here](https://github.com/brianchandotcom/liferay-portal/pull/157903#issuecomment-2603640586).

Also, in this [comment](https://liferay.slack.com/archives/C083CTA8Z3N/p1745348911460439?thread_ts=1744648647.681539&cid=C083CTA8Z3N), Shuyang Zhou mentioned that it would be ok to create a finder with performance in mind.

![image](https://github.com/user-attachments/assets/8840e110-fbf5-447f-b83a-ce3e48cfaa5e)